discovery.scaleway: port scaleway_sd_configs from Prometheus to Flow (#5101)
grafana/agent: docs/sources/flow/reference/components/discovery.scaleway.md

[baremetal services][baremetal].

[instance]: https://www.scaleway.com/en/virtual-instances/
[baremetal]: https://www.scaleway.com/en/bare-metal-servers/

## Usage

```river
discovery.scaleway "LABEL" {
    project_id = "SCALEWAY_PROJECT_ID"
    role       = "SCALEWAY_PROJECT_ROLE"
    access_key = "SCALEWAY_ACCESS_KEY"
    secret_key = "SCALEWAY_SECRET_KEY"
}
```

## Arguments

The following arguments are supported:

Name | Type | Description | Default | Required
---- | ---- | ----------- | ------- | --------
`project_id` | `string` | Scaleway project ID of targets. | | yes
`role` | `string` | Role of targets to retrieve. | | yes
`api_url` | `string` | Scaleway API URL. | `"https://api.scaleway.com"` | no
`zone` | `string` | Availability zone of targets. | `"fr-par-1"` | no
`access_key` | `string` | Access key for the Scaleway API. | | yes
`secret_key` | `secret` | Secret key for the Scaleway API. | | conditional
`secret_key_file` | `string` | Path to file containing secret key for the Scaleway API. | | conditional
`name_filter` | `string` | Name filter to apply against the listing request. | | no
`tags_filter` | `list(string)` | List of tags to search for. | | no

> `refresh_interval` | `duration` | Frequency to rediscover targets. | | no

Reviewer: ```suggestion
`refresh_interval` | `duration` | Frequency to rediscover targets. | `"60s"` | no
```

> `port` | `number` | Default port on servers to associate with generated targets. | | no

Reviewer: ```suggestion
`port` | `number` | Default port on servers to associate with generated targets. | `80` | no
```

* `__meta_scaleway_instance_tags`: The list of tags associated with the server concatenated with a `,`.
* `__meta_scaleway_instance_type`: Commercial type of the server.
* `__meta_scaleway_instance_zone`: Availability zone of the server.

## Component health

`discovery.scaleway` is only reported as unhealthy when given an invalid
configuration. In those cases, exported fields retain their last healthy
values.

## Debug information

`discovery.scaleway` does not expose any component-specific debug information.

### Debug metrics

`discovery.scaleway` does not expose any component-specific debug metrics.

## Example

```river
discovery.scaleway "example" {
    project_id = "SCALEWAY_PROJECT_ID"
    role       = "SCALEWAY_PROJECT_ROLE"
    access_key = "SCALEWAY_ACCESS_KEY"
    secret_key = "SCALEWAY_SECRET_KEY"
}

prometheus.scrape "demo" {
    targets    = discovery.scaleway.example.targets
    forward_to = [prometheus.remote_write.demo.receiver]
}

prometheus.remote_write "demo" {
    endpoint {
        url = PROMETHEUS_REMOTE_WRITE_URL

        basic_auth {
            username = USERNAME
            password = PASSWORD
        }
    }
}
```

Replace the following:

> * Replace `SCALEWAY_PROJECT_ID` with the project ID of your Scaleway machines.
> * Replace `SCALEWAY_PROJECT_ROLE` with `baremetal` to discover [baremetal][]
>   machines or `instance` to discover [virtual instances][instance].
> * Replace `SCALEWAY_ACCESS_KEY` with your Scaleway API access key.
> * Replace `SCALEWAY_SECRET_KEY` with your Scaleway API secret key.
> * Replace `PROMETHEUS_REMOTE_WRITE_URL` with the URL of the Prometheus
>   remote_write-compatible server to send metrics to.
> * Replace `USERNAME` with the basic authentication username for the
>   remote_write API.
> * Replace `PASSWORD` with the basic authentication password for the

Reviewer 2: ```suggestion
* `SCALEWAY_PROJECT_ID`: The project ID of your Scaleway machines.
* `SCALEWAY_PROJECT_ROLE`: Set to `baremetal` to discover [baremetal][] machines or `instance` to discover [virtual instances][instance].
* `SCALEWAY_ACCESS_KEY`: Your Scaleway API access key.
* `SCALEWAY_SECRET_KEY`: Your Scaleway API secret key.
* `PROMETHEUS_REMOTE_WRITE_URL`: The URL of the Prometheus remote_write-compatible server to send metrics to.
* `USERNAME`: The username to use for authentication to the remote_write API.
* `PASSWORD`: The password to use for authentication to the remote_write API.
```

Updates to align with style used in other reference docs.
Added missing text.
